Add profanity profile

adrianlshaw opened the pull request:
My profile for profanity. Comments welcome.

SkewedZeppelin commented on etc/profanity.profile:
```diff
@@ -0,0 +1,44 @@
+# This file is overwritten after every install/update
+quiet
+# Persistent local customizations
+include unzip.local
```
unzip?

SkewedZeppelin commented on etc/profanity.profile:
```diff
@@ -0,0 +1,44 @@
+# Persistent global definitions
+include globals.local
+
+ignore net none
```
rm?

SkewedZeppelin commented on etc/profanity.profile:
```diff
@@ -0,0 +1,44 @@
+include disable-programs.inc
+include disable-xdg.inc
+
+mkdir ${HOME}/.config/profanity
```
mkdir is only needed before using a whitelist arg

SkewedZeppelin commented on etc/profanity.profile:
```diff
@@ -0,0 +1,44 @@
+private-cache
+private-dev
+private-tmp
+private-etc alternatives,localtime,mime.types,resolv.conf,ssl
```
sort etc dev

SkewedZeppelin commented on etc/profanity.profile:
```diff
@@ -0,0 +1,44 @@
+memory-deny-write-execute
+noexec ${HOME}
+noexec /tmp
+
```
trailing line

SkewedZeppelin commented on etc/profanity.profile:
```diff
@@ -0,0 +1,44 @@
+ignore net none
+
+include disable-common.inc
+include disable-devel.inc
```
add disable-exec.inc and disable-interpreters.inc

SkewedZeppelin commented on etc/profanity.profile:
```diff
@@ -0,0 +1,44 @@
+mkdir ${HOME}/.local/share/profanity
+noblacklist ${HOME}/.config/profanity
+noblacklist ${HOME}/.local/share/profanity
+
```
include whitelist-usr-share-common.inc and whitelist-var-common.inc

SkewedZeppelin commented on etc/profanity.profile:
```diff
@@ -0,0 +1,44 @@
+# Firejail profile for profanity
+# Description: profanity is an XMPP-OTR chat client for the terminal
```
profanity has omemo support too :)

SkewedZeppelin commented on etc/profanity.profile:
```diff
@@ -0,0 +1,44 @@
+
+caps.drop all
+netfilter
+nodvd
```
add no3d and nodbus

SkewedZeppelin commented on etc/profanity.profile:
```diff
@@ -0,0 +1,44 @@
+nodvd
+nogroups
+nonewprivs
+noroot
```
add nosound?

SkewedZeppelin commented on etc/profanity.profile:
```diff
@@ -0,0 +1,44 @@
+noroot
+notv
+nou2f
+protocol unix,inet,inet6
```
add novideo

SkewedZeppelin commented on etc/profanity.profile:
```diff
@@ -0,0 +1,44 @@
+private-etc alternatives,localtime,mime.types,resolv.conf,ssl
+
+memory-deny-write-execute
+noexec ${HOME}
```
replaced with disable-exec.inc

adrianlshaw commented:
Thanks for the review @SkewedZeppelin. Initially this started out as a whitelist, but then I became confused given the structure of existing profiles and the default template. I've addressed all of your comments aside from `disable-interpreters`, because that causes the program to not even start.

Any pro tip on how I can go about effectively debugging the root cause?

SkewedZeppelin commented:
:+1: 

> not even start

try adding the following above the other noblacklists
```
# Allow python (blacklisted by disable-interpreters.inc)
#include allow-python2.inc
include allow-python3.inc
```

SkewedZeppelin commented on etc/profanity.profile:
```diff
@@ -0,0 +1,45 @@
+include disable-programs.inc
+include disable-xdg.inc
+
+include whitelist-usr-share-common.inc
```
nitpick: this should be after the noblacklists and before the general options

adrianlshaw commented:
Great advice, thank you! I'll be more careful when submitting future profiles.

SkewedZeppelin commented:
Merged, thanks!
